a3fe/configuration/engine_config.py

```diff
@@ -5,298 +5,243 @@
+        data = info.data
+        if not ('timestep' in data and 'nmoves' in data):
+            return v
+        
+        time_per_cycle = _Decimal(str(data['timestep'])) * _Decimal(str(data['nmoves'])) / _Decimal('1000000')
+        runtime_decimal = _Decimal(str(v))
+        
+        if runtime_decimal % time_per_cycle != 0:
             raise ValueError(
                 f"Runtime must be a multiple of the time per cycle. "
-                f"Runtime: {self.runtime} ns, Time per cycle: {time_per_cycle} ns."
+                f"Runtime: {v} ns, Time per cycle: {float(time_per_cycle)} ns"
             )
-        
-        # Only try to modify the config file if it exists
-        cfg_file = _os.path.join(self.input_dir, "somd.cfg")
-        if _os.path.exists(cfg_file):
-            # Need to modify the config file to set the correction n_cycles
-            n_cycles = self._calculate_ncycles(self.runtime, time_per_cycle)
-            self._set_n_cycles(n_cycles)
-            print(f"Updated ncycles to {n_cycles} in somd.cfg")
+        return float(v)
 
-    constraint: str = _Field("hbonds", description="Constraint type, must be hbonds or all-bonds")
+    @_model_validator(mode="after")
+    def _check_cutoff_values(self):
+        """
+        Issue warnings if the user supplies certain contradictory or unusual combos.
+        """
+        cutoff_type = self.cutoff_type
+        cutoff_distance = self.cutoff_distance
+        rfd = self.reaction_field_dielectric
+
+        # 1) Only warn if user set reaction_field_dielectric != 78.3
+        if cutoff_type == "cutoffperiodic" and rfd != 78.3:
+            self._logger.warning(
+                "You have cutoff_type=cutoffperiodic but set a reaction_field_dielectric. "
+                f"This value ({rfd}) will be ignored by the engine."
+            )
+
+        # 2) Only warn if user picks e.g. cutoff_distance < 6 or > 18
+        if cutoff_type == "PME" and not (6.0 <= cutoff_distance <= 18.0):
+            self._logger.warning(
+                f"For PME, we recommend cutoff_distance in [6.0, 18.0], but you have {cutoff_distance}. "
+                "we'll still accept it."
+            )
+        return self
+    
+    @_model_validator(mode="after")
+    def _check_charge_difference(self):
+        if self.charge_difference != 0 and self.cutoff_type != "PME":
+            raise ValueError("Charge difference is non-zero but cutoff type is not PME.")
+        return self
 
-    @_validator('constraint')
+    @_field_validator('constraint')
     def _check_constraint(cls, v):
-        if v not in ['hbonds', 'all-bonds']:
-            raise ValueError('constraint must be hbonds or all-bonds')
+        if v not in ['hbonds', 'allbonds']:
+            raise ValueError('constraint must be hbonds or allbonds')
+        return v    
+
+    @_field_validator('hydrogen_mass_factor')
+    def _check_hmf_range(cls, v):
```

Comment:
Again please check in the field with gt and lt - simpler than a separate validator.